src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -496,10 +513,12 @@ public void ProcessPushNotification(PushNotification pushNotification, TimeSpan?
         private async Task InitializeAsync(bool ignoreFailures, IEnumerable<ConfigurationClient> availableClients, CancellationToken cancellationToken = default)
         {
             Dictionary<string, ConfigurationSetting> data = null;
+            Dictionary<string, ConfigurationSetting> dataCopy = null;
 
             try
             {
                 data = await ExecuteWithFailOverPolicyAsync(availableClients, (client) => LoadAll(client, cancellationToken), cancellationToken).ConfigureAwait(false);
+                dataCopy = data.ToDictionary(kvp => kvp.Key, kvp => new ConfigurationSetting(kvp.Value.Key, kvp.Value.Value, kvp.Value.Label, kvp.Value.ETag));

```

Comment:
`dataCopy` is only used to initialize `_applicationSettings`.  Can we do it without creating this new dictionary?